tests/integration/security/sds_egress/sds_istio_mutual_egress_test.go

```diff
@@ -0,0 +1,220 @@
+					tlsSuffix: istioMutualTLSSuffix,
+					response:  response.StatusCodeOK,
+				},
+				"SIMPLE TLS mode requests are routed through gateway but fail with 503": {
+					tlsSuffix: simpleTLSSuffix,
+					response:  response.StatusCodeUnavailable,
+				},
+			}
+
+			for name, tc := range testCases {
+				ctx.NewSubTest(name).
+					Run(func(ctx framework.TestContext) {
+						doIstioMutualTest(t, ctx, ns, tc.tlsSuffix, tc.response)
+					})
+			}
+		})
+}
+
+func doIstioMutualTest(
+	t *testing.T, ctx resource.Context, ns namespace.Instance, tlsSuffix, expectedResp string) {
+	var client echo.Instance
+	echoboot.NewBuilderOrFail(t, ctx).
+		With(&client, util.EchoConfig("client", ns, false, nil, g, p)).
+		BuildOrFail(t)
+	applyGatewayConfig(t, ns, tlsSuffix)
+	defer deleteGatewayConfig(t, ns, tlsSuffix)
+
+	// give the configuration a moment to kick in
+	time.Sleep(time.Second * 30)
+	pretestReqCount := getEgressRequestCountOrFail(t, ns, prom)
+
+	for i := 0; i < externalReqCount; i++ {
+		w := client.WorkloadsOrFail(t)[0]
+		responses, err := w.ForwardEcho(context.TODO(), &epb.ForwardEchoRequest{
+			Url:   externalURL,
+			Count: 1,
+		})
+		if err != nil {
+			t.Fatalf("failed to make request from echo instance to %s: %v", externalURL, err)
+		}
+		if int32(len(responses)) < 1 {
+			t.Fatalf("received no responses from request to %s", externalURL)
+		}
+		resp := responses[0]
+
+		if expectedResp != resp.Code {
+			t.Errorf("expected status %s but got %s", expectedResp, resp.Code)
+		}
+
+	}
+
+	// give prometheus some time to ingest the metrics
+	time.Sleep(time.Second * 10)
+	posttestReqCount := getEgressRequestCountOrFail(t, ns, prom)
+	newGatewayReqs := posttestReqCount - pretestReqCount
+	if newGatewayReqs != externalReqCount {
+		t.Errorf("expected %d requests routed through egress, got %d",
+			externalReqCount, newGatewayReqs)
+	}
+}
+
+// sets up the destination rule to route through egress, virtual service, and service entry
+func applySetupConfig(t *testing.T, ns namespace.Instance) {
+	t.Helper()
+
+	configFiles := []string{
+		"testdata/destination-rule-cnn.yaml",
+		"testdata/rule-route-sidecar-to-egress-cnn.yaml",
+		"testdata/service-entry-cnn.yaml",
+	}
+
+	for _, c := range configFiles {
+		if err := g.ApplyConfig(ns, file.AsStringOrFail(t, c)); err != nil {
+			t.Fatalf("failed to apply configuration file %s; err: %v", c, err)
+		}
+	}
+}
+
+func applyGatewayConfig(t *testing.T, ns namespace.Instance, suffix string) {
+	t.Helper()
+
+	configYaml := fmt.Sprintf("%s\n%s",
+		gatewayConfig, suffix)
+	t.Logf("applying gateway yaml: %s", configYaml)
+	g.ApplyConfigOrFail(t, ns, configYaml)
+}
+
+func deleteGatewayConfig(t *testing.T, ns namespace.Instance, suffix string) {
+	t.Helper()
+
+	configYaml := fmt.Sprintf("%s\n%s",
+		gatewayConfig, suffix)
+	g.DeleteConfigOrFail(t, ns, configYaml)
+}
+
+func getMetric(t *testing.T, prometheus prometheus.Instance, query string) (float64, error) {
```

Comment:
this seems to be a common test facility function, not exists anywhere that can be reused directly?

Reply:
Exists in `util.go` from the mixer tests... should probably move into common testing pkg